### PR TITLE
 ROX-18679: containerize integration tests

### PR DESCRIPTION
**Pull request by Stringy**
## Description

Containerize the integration tests, to bundle common dependencies and files. Now that we're running on the test host, containerizing the tests will make it easier to create test bundles and traceable artifacts for each build/test run.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

Images built locally, but relying on CI for additional testing of multi-arch builds and end-to-end use of the resulting images.

### CODE REVIEW

File: .github/workflows/collector-builder.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ on:
 
 env:
   COLLECTOR_TAG: ${{ inputs.collector-tag }}
-  DEFAULT_BUILDER_TAG: master
+  DEFAULT_BUILDER_TAG: cache
```
Comment (Stringy) [inline .github/workflows/collector-builder.yml:18]: - [ ] don't forget to revert this change

File: ansible/roles/run-test-target/tasks/test-collection-method.yml [hunk 2]
```diff
@@ -32,19 +33,55 @@
...
 - block:
+
+    # Some considerations for running the test container:
+    #   1. the runtime auth must be mounted to allow the tests
+    #      to pull images. (~/.docker/config.json)
+    #   2. /tmp is mounted to allow a common mount point for
+    #      communication between containers (see listening ports test for
+    #      an example.
+    #
+    # One or both of these should be improved in the future. Moving auth
```
[the remark this answers is not in the document]
Comment (Stringy) [reply ansible/roles/run-test-target/tasks/test-collection-method.yml:62]: Agreed, and your suggestion about having the image authenticate itself is what I was trying to convey in this comment - I'll tweak it a bit so it's clearer.

File: integration-tests/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
+FROM --platform=$BUILDPLATFORM golang:1.19 as builder
+
+ARG TEST_ROOT="/tests"
```
[the remark this answers is not in the document]
Comment (Stringy) [reply integration-tests/Dockerfile:3]: This didn't quite work unfortunately, but does mean that the value doesn't need to be redefined in each target (still have to have `ARG TEST_ROOT`)

**Review by erthalion — APPROVED** [review]
Looks good to me. One minor question due to my lack of golang details -- I see that the original version was building the test binary specifying GOARCH & GOOS, and now we don't do that. Does it impact anything?

**Comment by Stringy** [comment]
>  the test binary specifying GOARCH & GOOS, and now we don't do that

Since we're using buildx to build multi-arch images, all of this is handled by Go implicitly, at least in the sense that it's building a 'native' binary for each platform (via qemu)
